dashboard/pages/primary_screening/callbacks.py

```diff
@@ -0,0 +1,98 @@
+from dash import html, no_update, callback, Output, Input, State
+
+from ..error import throwable
+from ...data.bmg_plate import parse_bmg_files_from_iostring
+from ...data.file_preprocessing.echo_files_parser import EchoFilesParser
+import io
+import base64
+import functools
+import uuid
+
+
+def upload_bmg_data(contents, names, last_modified, stored_uuid, file_storage):
+    if contents is None:
+        return no_update
+
+    if not stored_uuid:
+        stored_uuid = str(uuid.uuid4())
+
+    bmg_files = []
+
+    for content, filename in zip(contents, names):
+        name, extension = filename.split(".")
+        if extension == "txt":
+            _, content_string = content.split(",")
+            decoded = base64.b64decode(content_string)
+            bmg_files.append((filename, io.StringIO(decoded.decode("utf-8"))))
+
+    if bmg_files:
+        bmg_df, val = parse_bmg_files_from_iostring(tuple(bmg_files))
+        serialized_processed_df = bmg_df.reset_index().to_parquet()
```

Comment:
Important: We need to also store plate values themselves for future use (here as `val`).

`val` will be an `np.ndarray` composed of all plates' values - to serialize a numpy array take a look [here](https://numpy.org/doc/stable/reference/generated/numpy.savez_compressed.html)